Tasks/NuGetCommand/nugetpublisher.ts

```diff
@@ -0,0 +1,340 @@
+                    settings: vstsNuGetPushSettings
+                }
+
+                for (const packageFile of filesList) {
+                    await publishPackageVstsNuGetPushAsync(packageFile, publishOptions);
+                }
+            }
+            else {
+                tl.debug('Using NuGet.exe to push the packages');
+                let publishOptions = new PublishOptions(
+                    nuGetPath,
+                    feedUri,
+                    apiKey,
+                    configFile,
+                    verbosity,
+                    authInfo,
+                    environmentSettings);
+
+                for (const packageFile of filesList) {
+
+                    await publishPackageNuGetAsync(packageFile, publishOptions, authInfo);
+                }
+            }
+
+        } finally {
+            credCleanup();
+        }
+
+        tl.setResult(tl.TaskResult.Succeeded, tl.loc("PackagesPublishedSuccessfully"));
+
+    } catch (err) {
+        tl.error(err);
+
+        if (buildIdentityDisplayName || buildIdentityAccount) {
+            tl.warning(tl.loc("BuildIdentityPermissionsHint", buildIdentityDisplayName, buildIdentityAccount));
+        }
+
+        tl.setResult(tl.TaskResult.Failed, tl.loc("PackagesFailedToPublish"));
+    }
+}
+
+function publishPackageNuGetAsync(packageFile: string, options: PublishOptions, authInfo: auth.NuGetAuthInfo): Q.Promise<number> {
+    let nugetTool = ngToolRunner.createNuGetToolRunner(options.nuGetPath, options.environment, authInfo);
+    nugetTool.arg("push");
+
+    nugetTool.arg(packageFile);
+    
+    nugetTool.arg("-NonInteractive");
+
+    nugetTool.arg(["-Source", options.feedUri]);
+
+    nugetTool.argIf(options.apiKey, ["-ApiKey", options.apiKey]);
+
+    if (options.configFile) {
+        nugetTool.arg("-ConfigFile");
+        nugetTool.arg(options.configFile);
+    }
+
+    if (options.verbosity && options.verbosity !== "-") {
+        nugetTool.arg("-Verbosity");
+        nugetTool.arg(options.verbosity);
+    }
+
+    return nugetTool.exec();
+}
+
+async function publishPackageVstsNuGetPushAsync(packageFile: string, options: VstsNuGetPushOptions) {
+    let vstsNuGetPushTool = vstsNuGetPushToolRunner.createVstsNuGetPushToolRunner(options.vstsNuGetPushPath, options.settings, options.internalAuthInfo);
+    vstsNuGetPushTool.arg(packageFile);
+    vstsNuGetPushTool.arg(["-Source", options.feedUri]);
+    vstsNuGetPushTool.arg(["-AccessToken", options.internalAuthInfo.accessToken]);
+    vstsNuGetPushTool.arg("-NonInteractive")
+
+    if (options.verbosity && options.verbosity.toLowerCase() === "detailed") {
+        vstsNuGetPushTool.arg(["-Verbosity", "Detailed"]);
+    }
+
+    let exitCode: number = await vstsNuGetPushTool.exec();
+    if (exitCode === 0)
+    {
+        return;
+    }
+
+    // ExitCode 2 means a push conflict occurred
+    if (exitCode === 2 && options.settings.continueOnConflict)
+    {
+        return;
```

Comment:
maybe a debug message that says I'm ignoring the conflict since the user asked me to.